File: CS/GridControlCRUD.Common/DataModel/DataProviderAsyncExtensions.cs
using DevExpress.CRUD.DataModel;
using DevExpress.Xpf.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Threading.Tasks;

namespace DevExpress.CRUD.ViewModel {
    public static class DataProviderAsyncExtensions {
        public static async Task<IList<T>> ReadAsync<T>(this IDataProvider<T> dataProvider) where T : class {
#if DEBUG
            await Task.Delay(500);
#endif
            return await Task.Run(() => dataProvider.Read());
        }
        public static async Task<TResult> GetQueryableResultAsync<T, TResult>(this IDataProvider<T> dataProvider, Func<IQueryable<T>, TResult> getResult) where T : class {
#if DEBUG
            await Task.Delay(500);
#endif
            return await Task.Run(() => dataProvider.GetQueryableResult(getResult));
        }
        public static async Task UpdateAsync<T>(this IDataProvider<T> dataProvider, T entity) where T : class {
#if DEBUG
            await Task.Delay(500);
#endif
            await Task.Run(() => dataProvider.Update(entity));
        }
        public static async Task CreateAsync<T>(this IDataProvider<T> dataProvider, T entity) where T : class {
#if DEBUG
            await Task.Delay(500);
#endif
            await Task.Run(() => dataProvider.Create(entity));
        }
    }
}

File: VB/GridControlCRUD.Common/DataModel - VB/DataProviderAsyncExtensions.vb
Imports DevExpress.CRUD.DataModel
Imports DevExpress.Xpf.Data
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Linq.Expressions
Imports System.Threading.Tasks

Namespace DevExpress.CRUD.ViewModel
	Public Module DataProviderAsyncExtensions
		<System.Runtime.CompilerServices.Extension> _
		Public Async Function ReadAsync(Of T As Class)(ByVal dataProvider As IDataProvider(Of T)) As Task(Of IList(Of T))
#If DEBUG Then
			Await Task.Delay(500)
#End If
			Return Await Task.Run(Function() dataProvider.Read())
		End Function
		<System.Runtime.CompilerServices.Extension> _
		Public Async Function GetQueryableResultAsync(Of T As Class, TResult)(ByVal dataProvider As IDataProvider(Of T), ByVal getResult As Func(Of IQueryable(Of T), TResult)) As Task(Of TResult)
#If DEBUG Then
			Await Task.Delay(500)
#End If
			Return Await Task.Run(Function() dataProvider.GetQueryableResult(getResult))
		End Function
		<System.Runtime.CompilerServices.Extension> _
		Public Async Function UpdateAsync(Of T As Class)(ByVal dataProvider As IDataProvider(Of T), ByVal entity As T) As Task
#If DEBUG Then
			Await Task.Delay(500)
#End If
			Await Task.Run(Sub() dataProvider.Update(entity))
		End Function
		<System.Runtime.CompilerServices.Extension> _
		Public Async Function CreateAsync(Of T As Class)(ByVal dataProvider As IDataProvider(Of T), ByVal entity As T) As Task
#If DEBUG Then
			Await Task.Delay(500)
#End If
			Await Task.Run(Sub() dataProvider.Create(entity))
		End Function
	End Module
End Namespace
